apps/src/music/views/SoundsPanel2.tsx

```diff
@@ -0,0 +1,487 @@
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import classNames from 'classnames';
+import {getBaseAssetUrl} from '../appConfig';
+import styles from './soundsPanel2.module.scss';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon/FontAwesomeV6Icon';
+import MusicLibrary, {
+  SoundData,
+  SoundFolder,
+  SoundType,
+} from '../player/MusicLibrary';
+import SoundStyle from '../utils/SoundStyle';
+import FocusLock from 'react-focus-lock';
+import SegmentedButtons from '@cdo/apps/componentLibrary/segmentedButtons';
+import musicI18n from '../locale';
+
+/*
+ * Renders a UI for previewing and choosing samples. This is currently used within a
+ * custom Blockly Field {@link FieldSounds}
+ */
+
+type Mode = 'packs' | 'sounds';
+type Filter = 'all' | SoundType;
+
+type SoundEntry = {
+  folder: SoundFolder;
+  sound: SoundData;
+};
+
+// FolderPanelRow: A single folder.
+
+interface FolderPanelRowProps {
+  libraryGroupPath: string;
+  playingPreview: string;
+  folder: SoundFolder;
+  currentValue: SoundFolder;
+  onSelect: (path: SoundFolder) => void;
+  onPreview: (path: string) => void;
+  currentFolderRefCallback: (ref: HTMLDivElement) => void;
+}
+
+const FolderPanelRow: React.FunctionComponent<FolderPanelRowProps> = ({
```

Comment:
Is `FolderPanelRow` identical between SoundsPanel and SoundsPanel2? If so, should we reuse?

Reply:
It has diverged a tiny bit now, and I have been anticipating a bit more to come.  I think it might be safest to keep it separate to minimize risk of regressing the earlier version.